[opentok-node] Add OpenTok v2.8 typings and tests

pronebird opened the pull request:
<!--- # Please remember to use `describe` and `it`in the tests! see https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md for details. --->

- Links to documentation: https://tokbox.com/developer/sdks/node/reference/OpenTok.html
- Link to GitHub or NPM: https://github.com/opentok/opentok-node
- Type of contribution: new definition

Other notes:

This is mainly a fork of  `definitions/npm/opentok_v2.6.x` which adds missing and newly added features in OpenTok v2.7 and v2.8.

pascalduez commented on definitions/npm/opentok_v2.8.x/flow_v0.61.x-/opentok_v2.8.x.js:
```diff
@@ -0,0 +1,137 @@
+  |};
+  declare export type StartBroadcastOptions = {|
+    outputs: {
+      hls?: Object,
```
`Object` is `any` from Flow 0.75 on. `{}` or `{ +[string]: mixed }` are alternatives.

pronebird replied:
Fixed this with `{}`.

pronebird commented:
Fixed a few tests and removed a test for `getArchive` in favor of combined `test Archive` test. Some minor wording fix in tests for `Broadcast` as well.
